app/demo-hifive-inventor/app.toml

```diff
@@ -30,6 +30,7 @@ interrupts = { "plic.irq" = 0x1 }
 ints = [52]
 tasks = ['"rtc"']
 notification = [1]
+source = ['"rtc"']
```

Comment:
Can we not use the task name itself? Do you envision cases where task and source are different?

Reply:
I did this to handle cases where a single task owns multiple external interrupts. There's no good way to give descriptive names without requiring the source be given as well. It is somewhat redundant in cases where there's just one interrupt source.

Reply:
Ah okay. That makes sense. So, in that case, would the configuration look something like this:
```
[tasks.ext_int_ctrl.config]
ints = [52]
tasks = ['"rtc", "rtc"']
notification = [1, 2]
source = ['"source-a", "source-b"']
```

Reply:
There would also have to be another entry in `ints`, but otherwise yes.

Reply:
Oops, yeah. Missed the entry in `ints`.

Reply:
Actually your example made me realize there's a possible issue of there being illegal characters in the names... do you think it's worth validating that? There are a lot of characters, and the compiler would detect those anyways when trying to build the generated code.

Reply:
How is similar validation handled for rest of the elements in the toml file? Is it mostly left up to the compiler to deal with it?

Reply:
Pretty much every other string in the `toml` is either used to lookup an entry in a `<String, {X}>` map or open a file, there's no other code generated based off of strings as far as I'm aware. In these cases they just do a `.unwrap()` or `.expect()` on the `Result`.

Reply:
Ah I see. If you think the validation is necessary here to throw proper errors, then let's do it. If the compiler gives an error which is easy to decode and understand what went wrong, we can rely on that instead.

Reply:
I just tried changing the `source` to `r-t-c` and got the following error:

```
error: expected one of `:`, `;`, or `=`, found `-`
 --> /hubris/target/riscv32imc-unknown-none-elf/release/build/drv-fe310-rtc-73eab97c1b4d4483/out/rtc_config.rs:1:8
  |
1 | const R-T-C_NOTIFICATION: u32 = 0x1;
  |        ^ expected one of `:`, `;`, or `=`
```

I think that gets the gist of the issue across, but it may not be immediately obvious how to fix this. I could add a scan of the string and accept only alphanumeric characters plus underscores.

Reply:
yeah, I think having a check in xtask would make it easier to decode the error or understand where exactly it is coming from.